dev/com.ibm.ws.security.javaeesec/src/com/ibm/ws/security/javaeesec/identitystore/IdentityStorePermissionCheck.java

```diff
@@ -0,0 +1,23 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package com.ibm.ws.security.javaeesec.identitystore;
+
+/**
+ * Interface to abstract the identity store permission check based on criteria
+ * such as Jakarta Securty version (i.e. 3.0 vs. 4.0+).
+ */
+interface IdentityStorePermissionCheck {
+
+    /**
+     * Ensure implementors of getCallerGroups() can check that the calling context
+     * has IdentityStorePermission.
+     */
+    void checkGetGroupsPermission();
```

Comment:
It would be better to be named `checkPermission(String name)` to allow for different names than get group permission.  Can also have one with `(String name, String action)` for completeness, but if we only ever use the single argument permission check, then it isn't needed.